### PR TITLE
New: Add read-only functionality to search

### PR DESCRIPTION
**Pull request by patrick-ivers**
#### Changes
- Add disabled styles to search component
- Prevent onClick and onChange events from firing when search is disabled

This functionality has been added to allow the TreePicker component in adslot-ui to be read-only.

#### Trello
https://trello.com/c/ciNHr1ys/5426-adslot-ui-add-support-for-read-only-treepicker

#### Screenshots
screen shot 2016-06-23 at 5 46 03 pm

### CODE REVIEW

File: src/components/alexandria/SearchComponent.js [hunk 1]
```diff
@@ -5,20 +5,26 @@ import React, { PropTypes } from 'react';
...
   onChange,
   onClear,
   placeholder,
   svgSymbolCancel,
   svgSymbolSearch,
   value,
 }) => {
-  const onChangeBound = (event) => onChange(_.get(event, 'target.value'));
+  const cancelClassSuffixes = !disabled ?
+    svgSymbolCancel.classSuffixes : ['gray-light'];
```
Comment (vinteo) [inline src/components/alexandria/SearchComponent.js]: I would just have these on one line

**Comment by vinteo** [comment]
LGTM

File: src/components/alexandria/SearchComponent.js [hunk 2]
```diff
@@ -5,20 +5,26 @@ import React, { PropTypes } from 'react';
...
+  const onChangeBound = !disabled ?
+    (event) => onChange(_.get(event, 'target.value')) : null;
+  const onClearClick = !disabled ? onClear : null;
```
Comment (omgaz) [inline src/components/alexandria/SearchComponent.js]: Click is an implementation of how we're choosing to implement onClear and `onClearClick` isn't giving us any indication of a modified or mapped state. What we're actually doing is mapping onClear to either null or our passed prop onClear. How about `mappedOnClear`?

File: src/styles/color.scss [hunk 3]
```diff
@@ -69,6 +69,7 @@ $color-background: $color-white;
 $color-link: $color-cyan;
 $color-link-hover: $color-cyan-dark;
 $color-text-placeholder: $color-gray;
+$color-text-disabled: $color-gray;
```
Comment (omgaz) [inline src/styles/color.scss]: 👍

File: src/components/alexandria/SearchComponent.js (hunk 1 above)
Comment (OndrejCholeva) [inline src/components/alexandria/SearchComponent.js]: `gray-light` -> `color-disabled`

File: test/components/alexandria/SearchComponentTest.js [hunk 4]
```diff
@@ -69,6 +80,16 @@ describe('SearchComponent', () => {
     expect(fireCount).to.equal(1);
   });
 
+  it('should not fire onClear when disabled and the user clicks the icon', () => {
+    let fireCount = 0;
+    const testFunction = () => { fireCount += 1; };
+
+    const component = shallow(<SearchComponent onClear={testFunction} value="a" disabled />);
+    const svgSymbolEl = component.find(SvgSymbolComponent);
```
Comment (omgaz) [inline test/components/alexandria/SearchComponentTest.js]: Could we not alternatively check that the svg icon doesn't have an onclick prop? Then we wouldn't need the firecountm wouldn't need to simulate the click... we can treat it as black box

File: src/components/alexandria/SearchComponent.js (hunk 1 above)
Comment (patrick-ivers) [inline src/components/alexandria/SearchComponent.js]: `color-disabled` is `color-gray-lighter`, which is different to the existing colour for the search icon, `color-gray-light`.

Comment (patrick-ivers) [inline src/components/alexandria/SearchComponent.js]: `color-disabled` is `color-gray-lighter`, which isn't the colour used in the designs. Should I make it `color-gray-lighter`? Either way, I agree it should be a semantic name.

Comment (OndrejCholeva) [inline src/components/alexandria/SearchComponent.js]: `color-icon-disabled` makes sense to me.

**Comment by vinteo** [comment]
Merging...

**Pull request merged**